src/map.rs

```diff
@@ -840,13 +834,13 @@ where
     K: Hash + Eq,
     S: BuildHasher,
 {
-    // FIXME: reduce duplication (compare with insert)
-    fn entry_phase_1<Sz>(&mut self, key: K) -> Entry<K, V>
+    fn probe_action<'a, Sz, A>(&'a mut self, key: K, action: A) -> A::Output
```

Comment:
Nice refactoring. I think this method needs a clearer name. We use "probe" in every loop when reading/writing to the map, so it is not specific enough. Even "insert phase 1" is an ok name - and `insert_phase_2` still exists, so it needs to be renamed in tandem, if we entirely remove anything named "phase 1".